common/workunit/workunit.cpp

```diff
@@ -6805,6 +6863,140 @@ const char *clusterTypeString(ClusterType clusterType, bool lcrSensitive)
     throwUnexpected();
 }
 
+bool extractFromWorkunitDAToken(const char * distributedAccessToken, StringBuffer * wuid, StringBuffer * user, StringBuffer * privKey)
+{
+    if (strncmp(distributedAccessToken, "HPCC[", 5))
+    {
+        DBGLOG("Not a valid workunit distributed access token");
+        return false;
+    }
+
+    const char * finger;
+    if (user)
+    {
+        finger = strstr(distributedAccessToken+5, "u=");
+        if (finger)
+        {
+            finger += 2;
+            while (*finger && *finger != ',' && *finger != ']')
+                user->append(1, (const char *)finger++);
+        }
+    }
+
+    if (wuid)
+    {
+        finger = strstr(distributedAccessToken+5, "w=");
+        if (finger)
+        {
+            finger += 2;
+            while (*finger && *finger != ',' && *finger != ']')
+                wuid->append(1, (const char *)finger++);
+        }
+    }
+
+    if (privKey)
+    {
+        finger = strstr(distributedAccessToken+5, "]");
+        if (finger)
+        {
+            ++finger;
+            while (*finger && *finger != ';')
+                privKey->append(1, (const char *)finger++);
+        }
+    }
+    return true;
+}
+
+//Verifies the given signed workunit distributed access token was created
+//and signed by the given wuid and user, and that the workunit is still active
+//Returns:
+//   0 : Success, token is valid and workunit is active
+//   1 : Signature does not verify (wuid/username don't match, or signature doesnt verify)
+//   2 : Workunit not active
+//   Throws if unable to open workunit
+CriticalSection verifyWorkunitDATokenCrit;
+int verifyWorkunitDAToken(const char * distributedAccessToken)
+{
+    CriticalBlock block(verifyWorkunitDATokenCrit);
+
+    //Extract user
+    StringBuffer user;
+    const char * finger = strstr(distributedAccessToken, "u=");
+    if (finger)
+    {
+        finger += 2;
+        while (*finger && *finger != ',' && *finger != ']')
+            user.append(1, (const char *)finger++);
+    }
+
+    //Extract wuid
+    StringBuffer wuid;
+    finger = strstr(distributedAccessToken, "w=");
```

Comment:
again would be nice if we had general token parsing code.. that didn't use strstr, but rather relied on the format and provide something like an iproperties of the attributes.